### PR TITLE
add github homepage PR components

### PR DESCRIPTION
**Pull request by kissmikijr**
image

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by kissmikijr** [comment]
[SC-7787]

File: .changeset/serious-fishes-roll.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/backstage-plugin-github-pull-requests': major
+---
+
+Add review requests and your pull requests hompage components
```
Comment (Xantier) [inline .changeset/serious-fishes-roll.md:5]:
```suggestion
Add review requests and your pull requests homepage components
```

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithuRepositoryData.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export const useGithuRepositoryData = (url: string) => {
+  const githubAuthApi = useApi(githubAuthApiRef);
+
+  return useAsync(async (): Promise<GithubRepositoryData> => {
+    const token = await githubAuthApi.getAccessToken(['repo']);
+
+    const response = await new Octokit({
+      auth: token,
+      baseUrl: 'https://api.github.com',
```
Comment (Xantier) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithuRepositoryData.tsx:29]: Did we use the backend integration github url value here or has this been hardcoded elsewhere also? This wouldn't work on Github server/enterprise whatever it's called with a custom URL.

File: .changeset/serious-fishes-roll.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/backstage-plugin-github-pull-requests': major
```
Comment (Xantier) [inline .changeset/serious-fishes-roll.md:2]: I think we might get away with minor here. There doesn't seem to be breaking changes.
Could be good to bump major for vanity reasons though...

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithubSearchPullRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export const useGithubSearchPullRequest = (query: string) => {
+  const githubAuthApi = useApi(githubAuthApiRef);
+
+  return useAsync(async (): Promise<GithubSearchPullRequestsDataItem[]> => {
+    const token = await githubAuthApi.getAccessToken(['repo']);
+
+    const pullRequestResponse: GetSearchPullRequestsResponseType =
+      await new Octokit({
+        auth: token,
+        baseUrl: 'https://api.github.com',
```
Comment (Xantier) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithubSearchPullRequest.tsx:33]: Here as well URL.
Check `packages/integration/config.d.ts:89` on Backstage repo. The URL value is exposed to FE so we can use it from the config directly and default to api.github.com if there's no value. 
More info: https://backstage.io/docs/integrations/github/locations

**Review by Xantier — COMMENTED** [review]
Looking excellent. Few minor tweaks that I can see. 
I'll let you decide on the bump type

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithuRepositoryData.tsx (hunk 2 above)
Comment (kissmikijr) [reply plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithuRepositoryData.tsx:29]: You are right, I think we should use the backend integration url value. I put this in for iterating quickly and left it there. Gonna fix it.

File: .changeset/serious-fishes-roll.md (hunk 3 above)
Comment (kissmikijr) [reply .changeset/serious-fishes-roll.md:2]: I agree, I wanted it to be a minor change, but missed it :D

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithubSearchPullRequest.tsx (hunk 4 above)
Comment (kissmikijr) [reply plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithubSearchPullRequest.tsx:33]: 👍🏿

**Review by Xantier — APPROVED** [review]

**Pull request merged**